controllers/storagecluster/reconcile.go

```diff
@@ -403,6 +404,19 @@ func (r *StorageClusterReconciler) reconcilePhases(
 		return reconcile.Result{}, nil
 	}
 
+	storageClusterPeerList := &ocsv1.StorageClusterList{}
+	err := r.Client.List(r.ctx, storageClusterPeerList, client.InNamespace(instance.Namespace))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list storageClusterPeer: %w", err)
+	}
+	for i := range storageClusterPeerList.Items {
+		scp := &storageClusterPeerList.Items[i]
+		err := controllerutil.SetControllerReference(instance, scp, r.Scheme)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to set controller reference on StorageClusterPeer %v: %w", scp.Name, err)
+		}
+	}
```

Comment:
You are setting the controller ref on an in-memory object but you never update k8s with the resource update.